packages/rum-recorder/src/domain/segmentCollection.ts

```diff
@@ -0,0 +1,138 @@
+import { CreationReason, MouseMoveRecord, Record, RecordType, SegmentContext, SegmentMeta } from '../types'
+import { getRecordStartEnd, groupMouseMoves, isMouseMoveRecord } from './recordUtils'
+
+export const MAX_SEGMENT_DURATION = 30_000
+export const MAX_MOUSE_MOVE_BATCH = 100
+
+export interface SegmentWriter {
+  write(data: string): void
+  complete(data: string, meta: SegmentMeta): void
+}
+
+export function startSegmentCollection(getSegmentContext: () => SegmentContext | undefined, writer: SegmentWriter) {
+  let currentSegment: Segment | undefined
+
+  renewSegment('init')
+
+  function renewSegment(creationReason: CreationReason) {
+    if (currentSegment) {
+      currentSegment.finish()
+      currentSegment = undefined
+    }
+
+    const context = getSegmentContext()
+    if (!context) {
+      return
+    }
+
+    const localSegment = (currentSegment = new Segment(writer, context, creationReason))
+
+    // Replace the newly created segment after MAX_SEGMENT_DURATION
+    setTimeout(() => {
+      if (currentSegment === localSegment) {
```

Comment:
It is [covered by unit tests](https://github.com/DataDog/browser-sdk/blob/benoit/implement-recorder/packages/rum-recorder/src/domain/segmentCollection.spec.ts#L61-L69). You are right, we could also clear the timeout on renew segment. Do you think it would make things clearer?

Reply:
After removing this check, tests are still successful. We should have a test that fail without it.
IMO, clearing the timeout seems less complex to grasp than having two competing segments.